File: gradle/packages/cli-function/src/main/kotlin/com/prog/cli/function/Greeter.kt
package com.prog.cli.function

import com.prog.goodbye.world.printGoodbye
import com.prog.hello.world.printGreeting

object Greeter {
    /**
     * # `main`
     *
     * This function is entry point of the application.
     */
    @JvmStatic
    fun main(args: Array<String>) {
        println("===================================================")
        printWelcomeMessage(args)
        printGoodbyeMessage(args)
        println("===================================================")
    }

    /**
     * # `printWelcomeMessage`
     *
     * This function prints the welcome message by calling the `printGreeting`
     * function.
     */
    fun printWelcomeMessage(args: Array<String>) {
        println(printGreeting(args))
    }

    /**
     * # `printGoodbyeMessage`
     *
     * This function prints the goodbye message by calling the `printGoodbye`
     * function.
     */
    fun printGoodbyeMessage(args: Array<String>) {
        System.err.println(printGoodbye(args))
    }
}

File: csharp/src/CLI.Function/Greeter/Greeter.cs
using Prog.Goodbye.World;
using Prog.Hello.World;

namespace Prog.CLI.Function;

public class Greeter
{
    /// <summary> <h1><c>PrintWelcomeMessage</c></h1>
    /// <br/><br/>This method prints the welcome message.<br/><br/>
    ///
    /// <b>Arguments:</b>
    ///
    /// <list type="bullet">
    /// <item>
    /// <c>args: string[]</c> - The command line arguments passed to the
    /// program.
    /// </item>
    /// </list>
    /// </summary>
    public static void PrintWelcomeMessage(string[] args)
    {
        Console.WriteLine(HelloWorld.PrintGreeting(args));
    }

    /// <summary> <h1><c>PrintGoodbyeMessage</c></h1>
    /// <br/><br/>This method prints the goodbye message.<br/><br/>
    ///
    /// <b>Arguments:</b>
    ///
    /// <list type="bullet">
    /// <item>
    /// <c>args: string[]</c> - The command line arguments passed to the
    /// program.
    /// </item>
    /// </list>
    /// </summary>
    public static void PrintGoodbyeMessage(string[] args)
    {
        Console.Error.WriteLine(GoodbyeWorld.PrintGoodbye(args));
    }
}
